WalletWasabi/Extensions/HttpResponseMessageExtensions.cs

```diff
@@ -86,7 +87,15 @@ public static async Task<string> ToHttpStringAsync(this HttpResponseMessage me)
 
 		public static async Task ThrowRequestExceptionFromContentAsync(this HttpResponseMessage me)
 		{
-			var error = await me.Content.ReadAsJsonAsync<string>().ConfigureAwait(false);
+			var error = "";
+			try
+			{
+				error = await me.Content.ReadAsJsonAsync<string>().ConfigureAwait(false);
```

Comment:
`error` is initialized to `""` and in case of parsing error we log the exception. The problem is that `errorMessage` will be `\n` instead of `""`. This is not a big problem btw.

Reply:
{StatusCode: 404, ReasonPhrase: 'Not Found', Version: 1.1, Content: System.Net.Http.ByteArrayContent, Headers:
{
  Date: Wed, 01 Jul 2020 12:39:38 GMT
  Server: Kestrel
  Content-Length: 0
}}

Reply:
jsonString: `"<html>\r\n<head><title>404 Not Found</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>404 Not Found</h1></center>\r\n<hr><center>nginx</center>\r\n</body>\r\n</html>\r\n"`

trace: https://github.com/molnard/WalletWasabi/blob/2feb77ce3ef6422c9f145840708c49a1add1b08f/WalletWasabi/Extensions/HttpContentExtensions.cs#L23

exception: `Newtonsoft.Json.JsonReaderException: 'Unexpected character encountered while parsing value: <. Path '', line 1, position 1.'
`

Reply:
Ok, I got it. The problem is because we are trying to parse an HTML even when we know the content is not an json. 

We know that something was Not Found but we do not know if the error is returned by ngnix or wasabi so, we do not know whether the 404 is an html or a json. However, the `Content-Type` header let us know that because wasabi will return a `content-type: application/json` while ngnix will not return that or will return `content-type: text/html`. 

I think we can check that.

Reply:
Kestrel is not setting the content-type the only Nginx does. So I am checking that only.